docs/src/main/java/sample/registration/custommetadata/CustomMetadataConfig.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sample.registration.custommetadata;
```

Comment:
I would prefer that we enhance the existing classes under `sample.registration` to reduce code duplication.
Please merge `CustomMetadataConfig` into `SecurityConfig` and enhance `ClientRegistrar`.